R/as_gt.R

```diff
@@ -303,178 +240,102 @@ as_gt.gs_design <- function(
-    title <- "Bound summary of Binary Endpoint"
-  }
+  if (is.null(title)) title <- paste("Bound summary", switch(
+    method,
+    ahr = "for AHR design", wlr = "for WLR design",
+    combo = "for MaxCombo design", rd = "of Binary Endpoint"
+  ))
 
   # set different default subtitle to different methods
-  if (method == "ahr" && is.null(subtitle)) {
-    subtitle <- "AHR approximations of ~HR at bound"
-  }
-  if (method == "wlr" && is.null(subtitle)) {
-    subtitle <- "WLR approximation of ~wHR at bound"
-  }
-  if (method == "combo" && is.null(subtitle)) {
-    subtitle <- "MaxCombo approximation"
-  }
-  if (method == "rd" && is.null(subtitle)) {
-    subtitle <- "measured by risk difference"
-  }
+  if (is.null(subtitle)) subtitle <- switch(
+    method,
+    ahr = "AHR approximations of ~HR at bound",
+    wlr = "WLR approximation of ~wHR at bound",
+    combo = "MaxCombo approximation",
+    rd = "measured by risk difference"
+  )
 
   # set different default columns to display
-  if (is.null(display_columns)) {
-    if (method == "ahr") {
-      display_columns <- c(
-        "Analysis", "Bound", "Z", "Nominal p",
-        "~HR at bound", "Alternate hypothesis", "Null hypothesis"
-      )
-    } else if (method == "wlr") {
-      display_columns <- c(
-        "Analysis", "Bound", "Z", "Nominal p",
-        "~wHR at bound", "Alternate hypothesis", "Null hypothesis"
-      )
-    } else if (method == "combo") {
-      display_columns <- c(
-        "Analysis", "Bound", "Z", "Nominal p",
-        "Alternate hypothesis", "Null hypothesis"
-      )
-    } else if (method == "rd") {
-      display_columns <- c(
-        "Analysis", "Bound", "Z", "Nominal p",
-        "~Risk difference at bound", "Alternate hypothesis", "Null hypothesis"
-      )
-    }
-  }
+  if (is.null(display_columns)) display_columns <- c(
+    "Analysis", "Bound", "Z", "Nominal p",
+    sprintf("%s at bound", switch(method, ahr = "~HR", wlr = "~wHR", rd = "~Risk difference")),
+    "Alternate hypothesis", "Null hypothesis"
+  )
   # filter the columns to display as the output
   ## if `Probability` is selected to output, then transform it to `c("Alternate hypothesis", "Null hypothesis")`
-  if ("Probability" %in% display_columns) {
-    display_columns <- display_columns[!display_columns == "Probability"]
-    display_columns <- c(display_columns, "Alternate hypothesis", "Null hypothesis")
-  }
+  if (any(i <- display_columns == "Probability"))
+    display_columns <- c(display_columns[!i], "Alternate hypothesis", "Null hypothesis")
   ## check if the `display_columns` are included in `x` output
-  if (sum(!(display_columns %in% names(x))) >= 1) {
-    stop("as_gt: the variable names in display_columns is not outputted in the summary_bound object!")
-  } else {
-    x <- x %>% dplyr::select(dplyr::all_of(display_columns))
-  }
+  if (!all(display_columns %in% names(x))) stop(
+    "not all variable names in 'display_columns' are in the summary_bound object!"
```

Comment:
Echo with my previous comment, shall we add `as_gt` at the beginning of the error message?